### ISSUE TITLE
Python not finding/compiling lots of "necessary bits".

### ISSUE THREAD

**Issue by erigler-usgs** [Linuxbrew/linuxbrew#875]
This is related to my own #715, as well as #460, and similar issues...

In my attempts to (re)install Python, I quickly noticed that the zlib extension was once again failing to compile (pip won't even run without it, which is pretty much a show-stopper for me). After several unsuccessful attempts to fix this by playing with dependencies, I broke down and looked closely at the setup.py script included in the Python package. This is where Python  attempts to determine which "necessary bits" for building various extensions are available by looking for associated files.

As a brief explanation, Python's setup.py looks for associated header files to determine if "necessary bits" are available. Sometimes it also tries to ensure that the associated library file is available. It does this by parsing the CLAGS and LDFLAGS variables (not sure why the CPPFLAGS fix in #460 ever worked, but I assume it was some kind of makefile magic), and then looking for the .h or .so file in the generated list of inc_dirs and lib_dirs. Anyway, the setup.py section related to zlib was clearly looking for libz in its long list of lib_dirs, and not finding it, so it added zlib to its list of missing "necessary bits", which prevented it from getting compiled into Python.

I was able to fix this by adding an appropriate value to ldflags in python.rb. What's more, several other previously missing "necessary bits" were found and compiled, including _ssl, bz2, _curses, readline, and gdbm. Python's setup.py also found _curses_panel, but failed when compiling, presumably because it couldn't find the panel.h file located in $(brew --prefix)/include/ncursesw/ (maybe this deserves a separate bug report, but I'm not currently motivated to teach myself what _curses_panel actually is...it's not something I use).

Long story short, it seems necessary to define CFLAGS and/or CPPFLAGS **AND LDFLAGS** in python.rb to get many of Python's "necessary bits" to compile for many of its more popular extensions.

**Comment by sjackman** [comment]
It looks as though the `python` formula totally ignores the `LDFLAGS` environment variable provided by https://github.com/Linuxbrew/linuxbrew/blob/master/Library/Homebrew/extend/ENV/std.rb#L42.

Are you using your system's GCC or the Homebrewed GCC? Please report the output of `brew config`. `brew install gcc` should I believe also resolve your issue.

Please submit a pull request similar to cd680d7281f1b59b9603853403f562bd70ac97d2 for `LDFLAGS`.

**Comment by erigler-usgs** [comment]
Apologies for leaving out so much pertinent info.

1) this is a fresh "standalone" Linuxbrew installation on a Scientific Linux 7 system (basically, CentOS);
2) I am using Linux-brewed GCC (v5.3.0), with all Linux-brewed glibc, binutils, etc.;
3) I do use a fake keg to access Xorg-related system libraries (a la #826);
3) Here is the output from `brew config`:

```
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/Linuxbrew/linuxbrew.git
HEAD: 84012dc6ebdace816a086db2c851c8b25b4eefd8
Last commit: 2 days ago
HOMEBREW_PREFIX: /home/erigler/local
HOMEBREW_REPOSITORY: /home/erigler/local/linuxbrew
HOMEBREW_CELLAR: /home/erigler/local/linuxbrew/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://linuxbrew.bintray.com
CPU: dual-core 64-bit ivybridge
Kernel: Linux 3.10.0-327.3.1.el7.x86_64 x86_64 GNU/Linux
OS: Scientific Linux release 7.2 (Nitrogen)
OS glibc: 2.17
OS gcc: 4.8.5
Linuxbrew glibc: 2.19
Linuxbrew gcc: 5.3.0
Clang: N/A
X11: 2.7.8 => /usr
System Ruby: 2.3.0 => /home/erigler/local/linuxbrew/Cellar/ruby/2.3.0/bin/ruby
Perl: /usr/bin/perl
Python: /home/erigler/local/bin/python => /home/erigler/local/linuxbrew/Cellar/python/2.7.11/bin/python2.7
Ruby: /home/erigler/local/bin/ruby => /home/erigler/local/linuxbrew/Cellar/ruby/2.3.0/bin/ruby
Java: openjdk version "1.8.0_71"
```

Perhaps it's already obvious to you, but the `configure` line generated by Linuxbrew contains the LDFLAGS value for `$(brew --prefix)/lib` when I set it in python.rb, and it does **NOT** contain this LDFLAGS value when I do not set it in python.rb. I can't speak to what is or is not picked up from std.rb.

Finally, I am a real novice at Github. Can you guide me, or point me to some hints, on how to submit a useful PR?

**Comment by sjackman** [comment]
If you like you can use the GitHub web interface to edit the file.
Go to https://github.com/Linuxbrew/linuxbrew/blob/master/Library/Formula/python.rb and click the "edit" icon. Make your changes, and it'll open a pull request. Model it after the one in cd680d7, but for `ldflags`.

**Comment by erigler-usgs** [comment]
I think I successfully submitted a pull request (#886), for a single commit: https://github.com/Linuxbrew/linuxbrew/commit/e7f5829f75263228c610e144dc99faeff851f621

**Comment by sjackman** [comment]
```
==> brew audit python
Error: 1 problem in 1 formula
python:
 * 132: Trailing whitespace was found
==> FAILED
```
You can use `brew audit --style python` to catch style issues such as these.

**Comment by sjackman** [comment]
Fixed by #886.
